src/App.js

```diff
@@ -0,0 +1,104 @@
+import './App.css';
+import Todo from "./components/Todo"
+import Form from "./components/Form"
+import FilterButton from './components/FilterButton';
+
+import React, { useEffect, useState } from "react";
+import { nanoid } from "nanoid";
+
+function App(props) {
+  
+  const [tasks, setTasks] = useState(props.tasks);
+  const [filter, setFilter] = useState("All");
+  
+  
+  function addTask(name) {
+    if (name==="") return;
+    let newTask = {
+      id: "task-"+nanoid(),
+      name: name,
+      completed: false
+    };
+    setTasks([...tasks, newTask]);
+  }
+
+  function toggleTaskCompleted(id) {
+    const updatedTasks = tasks.map(task=>{
+      if (id===task.id) {
+        return {...task, completed: !task.completed}
+      }
+      return task;
+    })
+    setTasks(updatedTasks);
+  }
+  
+  function deleteTask(id) {
+    console.log(id);
+    const remainingTasks = tasks.filter(task=>id!==task.id);
+    setTasks(remainingTasks);
+  }
+  
+  function editTask(id, newName) {
+    const updatedTasks = tasks.map(task=>{
+      if (id===task.id) {
+        return {...task, name: newName}
+      }
+      return task;
+    })
+    setTasks(updatedTasks);
+  }
+  
+  
+  //tasks.map(task=>{console.log(task.id+" "+task.completed)})
+  
+  const taskList = tasks.filter(props.FILTER_MAP[filter]).map(task => (
+    <Todo 
+    id={task.id} 
+    name={task.name} 
+    completed={task.completed} 
+    key={task.id}
+    toggleTaskCompleted={toggleTaskCompleted}
+    deleteTask={deleteTask}
+    editTask={editTask}
+    />
+    ));
+    
+    const filterList = props.FILTER_NAMES.map(name=>(
+    <FilterButton 
+    key={name} 
+    name={name}
+    isPressed={name === filter}
+    setFilter={setFilter}  
+    />
+    ));
+    
+    const countString = tasks.length + (tasks.length===1?" task ":" tasks ") + "remaining";
```

Comment:
it's recommended to use string templating if you're doing concatenation. it makes it clearer that the operation is concat than addition

Reply:
Thank you for the feedback!  I have implemented the changes you suggested :)